### PR TITLE
Add logging of class and method names into Logger

### LINKED ISSUE #22204: Add logging of class and method names into Logger
In test automation framework present hard-coded class and method names in log messages.
```Logger.debug('Dashboard.stopAndRemoveWorkspaceByUI "${workspaceName}"');```
This logic should be moved into Logger class and improve to create this information dynamycly. 

Possible solutions from @dmytro-ndp [1]:
```
function getCallerInfo(): string {
  const error = new Error();

  // Use regular expressions to parse the stack trace
  const stackTrace = error.stack || '';
  const classNameMatch = stackTrace.match(/at\s+(\S+)\./);
  const className = classNameMatch ? classNameMatch[1] : undefined;

  let methodNameMatch;
  if (className) {
    const methodNameRegex = new RegExp(`at ${className}\\.(\\S+)`);
    methodNameMatch = stackTrace.match(methodNameRegex);
  }

  const methodName = methodNameMatch ? methodNameMatch[1] : undefined;

  return `${className}.${methodName}`
}
```

[1] https://github.com/eclipse/che/pull/22201#discussion_r1189665731

### PR DESCRIPTION
**Pull request by nallikaea**
### What does this PR do?
resolve #22204
reformat code

### Screenshot/screencast of this PR

image

### What issues does this PR fix or reference?

resolve #22204

### How to test this PR?

You can check new logs in Empty Workspase GitHub Action

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by musienko-maxim — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
@nallikaea : it would be interesting to see how will error or warning message look like. Would you please share anyone?

**Comment by dmytro-ndp** [comment]
Thank you, @nallikaea

I had also triggered devspaces tests from this PR https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/e2e/job/basic/job/typescript-tests/11393/ and found lots of `Context.<anonymous> - ` prefixes. IMHO, it makes sense to filter `Context.<anonymous> - ` and leave just text message, e.g.:
`Context.<anonymous> - editor.getText(): get recommended extensions as text from editor, delete comments and parse to object.` 
to replace with 
`Editor.getText(): get recommended extensions as text from editor, delete comments and parse to object.`.

Also, am I understand correctly, that next method call can't be identified automatically and so should be written up explicitly in the code? :
- `ActivityBar().getViewControl('Extensions'))?.openView()`
- `Editor.getText()`

**Review by dmytro-ndp — APPROVED** [review]

**Comment by nallikaea** [comment]
@dmytro-ndp

-  [Context](https://mochajs.org/api/context.js.html) is mocha class, it means that Logger was called inside mochas test function:

```
test(`Test name`, async function(): Promise<void> {
      Logger.debug(`some message`);
```
_returns: Context.<anonimus> - some message_

- In case of hard-coded functions it because Logger was called before some function inside the test and says what gonna be next. It is impossible to put our Logger inside monaco-page-object library functions.

```
test(`Test name`, async function(): Promise<void> {
    Logger.debug(`EditorView().openEditor(${extensionsListFileName})`);
    const editor: TextEditor = await new EditorView().openEditor(extensionsListFileName) as TextEditor;
```
_returns: Context.<anonimus> - EditorView().openEditor(${extensionsListFileName})_

maybe we can delete it because monaco-page-object library has own logger and will tell us something in case of error.

File: tests/e2e/utils/Logger.ts [hunk 1]
```diff
@@ -10,82 +10,113 @@
...
+    for (let i: number = 0; i < messageIndentationLevel; i++) {
+      console.groupEnd();
     }
+  }
+
+  private static getCallerInfo(): string {
+    const e: Error = new Error();
+    const stack: string[] = e.stack ? e.stack.split('\n') : [];
+    // " at functionName ( ..." => "functionName"
+    return stack[3].replace(/^\s+at\s+(.+?)\s.+/g, '$1');
```
Comment (dmytro-ndp) [inline tests/e2e/utils/Logger.ts:116]: What about don't logging `Context.<anonymous>` and `Runner.<anonymous>` to make log cleaner? 
```
stack[3].replace(/^\s+at\s+(.+?)\s.+/g, '$1')
            .replace('Context.<anonymous>', '')
            .replace('Runner.<anonymous> ', '')
```

E.g.:
```
    ✔ Get recommended extensions list from extensions.json (10623ms)
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
     ✔ Open "Extensions" view section (1885ms)
       • Context.<anonymous> - Time for extensions installation TimeoutConstants.TS_COMMON_PLUGIN_TEST_TIMEOUT=30000
             ‣ DriverHelper.wait - (30000 milliseconds)
     ✔ Let extensions complete installation (30003ms)
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
           ▼ Context.<anonymous> - ActivityBar().getViewControl('Extensions'))?.openView(): open Extensions view.
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): get current section.
             ‣ DriverHelper.waitAllPresence - By(css selector, .name)
       • Context.<anonymous> - Check if {"publisher":"ms-python","name":"python"} are installed.
           ▼ Context.<anonymous> - extensionSection.findItem(python).
             ‣ DriverHelper.isVisible - By(xpath, //a[text()='Reload Required'])
           ▼ Context.<anonymous> - extensionsView?.getContent().getSections(): switch to marketplace section.
             ‣ DriverHelper.waitVisibility - By(css selector, .monaco-list-rows)
             ‣ DriverHelper.waitVisibility - element is located and is visible.
           ▼ Context.<anonymous> - marketplaceSection.getVisibleItems(): get all found items.
           ▼ Context.<anonymous> - Try to find extension published by ms-python.
           ▼ Context.<anonymous> - Extension was found: Black Formatter
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled()
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher?.isInstalled(): true.
           ▼ Context.<anonymous> - itemWithRightNameAndPublisher.manage(): get context menu.
     1) Check if extensions are installed and enabled
   [ERROR] Runner.<anonymous> - CheReporter runner.on.fail: Check if recommended extensions installed for Ansible Check if extensions are installed and enabled failed after 3346ms
             ‣ Runner.<anonymous> - FullTitle:Check if recommended extensions installed for Ansible Check if extensions are installed and enabled
             ‣ Runner.<anonymous> - FullTitleSanitized:Check_if_recommended_extensions_installed_for_Ansible_Check_if_extensions_are_installed_and_enabled
             ‣ Runner.<anonymous> - TestTitle:Check if extensions are installed and enabled
             ‣ Runner.<anonymous> - TestTitleSanitized:Check_if_extensions_are_installed_and_enabled
             ‣ DriverHelper.getDriver  
```

Comment (nallikaea) [reply tests/e2e/utils/Logger.ts:116]: https://github.com/eclipse/che/issues/22448

**Pull request merged**